### DISCUSSION TITLE
Executable file attribute not coming across \\wsl$ network share

### DISCUSSION THREAD

**Discussion by riverar** [microsoft/WSL#6649] [General] [answered]
# Environment

```none
Windows 10 10.0.19042.844
Ubuntu 20.04
Linux version 5.4.72-microsoft-standard-WSL2 (oe-user@oe-host) (gcc version 8.2.0 (GCC))
#1 SMP Wed Oct 28 23:40:43 UTC 2020
```

# Steps to reproduce

1. Install [QT Creator](https://download.qt.io/official_releases/qtcreator/4.14/4.14.1/qt-creator-opensource-windows-x86_64-4.14.1.exe) for Windows, take defaults
2. Open QT Creator and navigate to Tools > Options > Kits and click the Compilers tab
3. Click the Add > GCC > C++
4. Specify `\\wsl$\Ubuntu\usr\bin\base64` as the Compiler Path
5. Observe the path is highlighted in red
6. Observe tooltip on path indicates: `"The path [...] is not an executable file"`

![image](https://user-images.githubusercontent.com/475132/110168315-db79bc00-7dab-11eb-864b-1477d24f5344.png)

**WSL logs**: https://aka.ms/AAbe0sl

#  Expected behavior

QT Creator recognizes the file is executable and lets me continue configuring the compiler toolchain.

# Actual behavior

QT Creator blocks further configuration of the compiler toolchain as it does not believe the specified file is executable. It appears that information is being lost when traversing the `\\wsl$` share.

**Comment by DHowett** [comment]
Does Windows QT Creator have the capacity to execute ELF binaries?

Wouldn’t it need to have a feature where it runs them inside a translation layer or VM ... like WSL?

**Comment by DHowett** [comment]
Like... just because the executable has the +x bit doesn’t mean that arbitrary Windows applications can _execute them_.

**Comment by riverar** [comment]
@DHowett I'm configuring QT Creator with the goal of setting up remote build steps. Sounds like this is "as designed" and I should forget about it though.

**Comment by DHowett** [comment]
Ah, that’s curious. I wonder what path format their remote build configuration expects!

**Comment by riverar** [comment]
@DHowett Not sure, to be honest. Didn't get far and not intimately familiar with the software. I'm scrapping the idea and will just install it in WSLv2 directly and X11 forward it out of the VM. Thanks for making me rethink my plan 😅

What's your thoughts on this issue? Shall I close it? Too niche?

**Comment by Biswa96** [comment]
Possible options:
1. Qt Creator can call a ~backdoor~ backend in WSL side to build the job, just like VSCode or the `init` itself.
2. Qt Creator can give the job to `qmake` in WSL side then take back the built binaries in build folder.

**Comment by riverar** [comment]
So @therealkenc, why was this moved to discussions? (Also, why is the General category in Q&A format?)

**Reply by therealkenc** [reply]
Prior to the advent of the discussion forum, submissions were being tagged "external", but as often as not, discussion would continue in the closed issue (as opposed to being migrated to a third-party forum). Whether a given submission get migrated or closed depends somewhat on whether there has been any take-up (contrast chirping crickets). Here there was. Discussion of third-party products in the WSL discussion forum is perfectly okay, but most times you'll get better eyeballs in the third-party product's forum, where there are more people who are familiar with the product in question. When bringing up topic in third-party forums, it is usually a good play to frame the ask in terms of Real Linux behavior. Here that might be Qt Creator on Windows using a path to a Samba share on the Linux side. In this case, copying the file onto NTFS suffices to demonstrate the cited error (`The path [...] is not an executable file`) is Qt Creator behavior.

![image](https://user-images.githubusercontent.com/7049700/110389243-192f4c80-8019-11eb-8fe1-e51fa6f94083.png)

**Reply by riverar** [reply]
The executable bit demonstratively doesn't come across when accessing via share. It's unclear if this is a bug or not, as a Microsoft engineer has not commented to that effect.

Frankly, I'm growing tired of clashing with you--a non Microsoft employee--in this repository where you close issues that don't meet an undocumented rubric. So I'm going to take my toys and go home.

**Comment by craigloewen-msft** [comment, marked as answer]
From the discussion above, it seems like what you're doing isn't a supported use case for QTCreator as @DHowett and @therealkenc  mentioned. I agree with your point, and it would be my recommendation that you get an X server and run QT Creator in a Linux context to resolve this. We're working on adding GUI app support in the future which will make this a lot easier!
